pkg/kvstore/etcd.go

```diff
@@ -497,3 +497,44 @@ func (e *EtcdClient) Status() (string, error) {
 	}
 	return "Etcd: " + strings.Join(eps, "; "), err1
 }
+
+// StartWatch starts watching for changes in a prefix
+func (e *EtcdClient) StartWatch(w *Watcher) {
+	go func(w *Watcher) {
+		for {
+		recreateWatcher:
+			etcdWatch := e.cli.Watch(ctx.Background(), w.prefix, client.WithPrefix())
+
+			select {
+			case <-w.stopWatch:
+				return
+
+			case r, ok := <-etcdWatch:
+				if !ok {
+					goto recreateWatcher
```

Comment:
In the event of etcd not being reachable doesn't this create a infinite loop with 100% CPU consumption?